controller.py

```diff
@@ -83,249 +83,183 @@ def get_docbins(
-            )
-            doc_ock.post_embedding_failed(request.user_id, request.config_string)
-            message = f"Model {request.config_string} is not supported. Please contact the support."
-            notification.create(
-                request.project_id,
-                request.user_id,
-                message,
-                enums.Notification.ERROR.value,
-                enums.NotificationType.EMBEDDING_CREATION_FAILED.value,
-                True,
-            )
-            send_project_update(
-                request.project_id, f"notification_created:{request.user_id}", True
-            )
-            return status.HTTP_422_UNPROCESSABLE_ENTITY
-        except ValueError:
-            embedding.update_embedding_state_failed(
-                request.project_id,
-                embedding_id,
-                with_commit=True,
-            )
-            send_project_update(
-                request.project_id,
-                f"embedding:{embedding_id}:state:{enums.EmbeddingState.FAILED.value}",
-            )
-            doc_ock.post_embedding_failed(request.user_id, request.config_string)
-            message = f"Model {request.config_string} was deleted during the creation process."
-            notification.create(
-                request.project_id,
-                request.user_id,
-                message,
-                enums.Notification.ERROR.value,
-                enums.NotificationType.EMBEDDING_CREATION_FAILED.value,
-                True,
-            )
-            send_project_update(
-                request.project_id, f"notification_created:{request.user_id}", True
-            )
-            return status.HTTP_422_UNPROCESSABLE_ENTITY
+        embedder = get_embedder(
+            project_id, embedding_type, iso2_code, platform, model, api_token
+        )
 
         if not embedder:
-            embedding.update_embedding_state_failed(
-                request.project_id,
-                embedding_id,
-                with_commit=True,
-            )
-            send_project_update(
-                request.project_id,
-                f"embedding:{embedding_id}:state:{enums.EmbeddingState.FAILED.value}",
-            )
-            doc_ock.post_embedding_failed(request.user_id, request.config_string)
-            message = f"The data type {attribute_data_type} is currently not supported for embeddings. Please contact the support."
-            notification.create(
-                request.project_id,
-                request.user_id,
-                message,
-                enums.Notification.ERROR.value,
-                enums.NotificationType.EMBEDDING_CREATION_FAILED.value,
-                True,
-            )
-            send_project_update(
-                request.project_id, f"notification_created:{request.user_id}", True
-            )
-            raise Exception(message)
-    except HTTPError:
-        print(traceback.format_exc(), flush=True)
```

Comment:
i think we shouldn't remove the tracebacks since it really hard to debug if no log is shown on error

Reply:
added again